### PR TITLE
More granual timers in VMCBatched.  Fixed ccmake ENABLE_TIMERS issue.

### PR DESCRIPTION
**Pull request by PDoakORNL**

### CODE REVIEW

File: src/QMCDrivers/QMCDriverNew.h [hunk 1]
```diff
@@ -85,6 +85,32 @@ class QMCDriverNew : public QMCDriverInterface, public MPIObjectBase
    */
   std::bitset<QMC_MODE_MAX> qmc_driver_mode_;
 
+  /** The timers for the driver.
+   *
+   * This cleans up the driver constructor, and a reference to this structure 
+   * Takes the timers into thread scope.
+   */
+  struct Timers
```
Comment (ye-luo) [inline src/QMCDrivers/QMCDriverNew.h]: Is DriverTimers a better name?

Comment (PDoakORNL) [reply src/QMCDrivers/QMCDriverNew.h]: Thinking about this even though its a nested type its less readable used as a argument to the threaded functions.  I'll change to `DriverTimers`

File: src/QMCDrivers/QMCDriverNew.h [hunk 2]
```diff
@@ -304,7 +331,10 @@ class QMCDriverNew : public QMCDriverInterface, public MPIObjectBase
 
   /** }@ */
 
-        std::vector<RandomGenerator_t*> RngCompatibility;
+  std::vector<RandomGenerator_t*> RngCompatibility;
+
+  Timers timers_;
```
Comment (ye-luo) [inline src/QMCDrivers/QMCDriverNew.h]: Is it under `public:`? I think it should be in protected.

File: src/QMCDrivers/QMCDriverNew.h (hunk 1 above)
Comment (PDoakORNL) [reply src/QMCDrivers/QMCDriverNew.h]: Done

File: src/QMCDrivers/QMCDriverNew.h (hunk 2 above)
Comment (PDoakORNL) [reply src/QMCDrivers/QMCDriverNew.h]: Done

**Comment by markdewing** [comment]
@ye-luo I will look into using ScopedTimer with a reference.

**Pull request merged**